src/ssh.c

```diff
@@ -1657,42 +1707,71 @@ static int DoSshPubKey(const byte* in, word32 inSz, byte** out,
 
 static int DoAsn1Key(const byte* in, word32 inSz, byte** out,
         word32* outSz, const byte** outType, word32* outTypeSz,
-        void* heap)
+        int isPrivate, void* heap)
 {
     int ret = WS_SUCCESS;
     byte* newKey = NULL;
+    WS_KeySignature* key = NULL;
 
     WOLFSSH_UNUSED(heap);
 
-    if (*out == NULL) {
-        newKey = (byte*)WMALLOC(inSz, heap, DYNTYPE_PRIVKEY);
-        if (newKey == NULL) {
-            return WS_MEMORY_E;
-        }
-    }
-    else {
-        if (*outSz < inSz) {
-            WLOG(WS_LOG_DEBUG, "DER private key output size too small");
-            return WS_BUFFER_E;
-        }
-        newKey = *out;
-    }
-
-    ret = IdentifyAsn1Key(in, inSz, 1, heap);
-
+    ret = IdentifyAsn1Key(in, inSz, isPrivate, heap, &key);
     if (ret > 0) {
-        *out = newKey;
-        *outSz = inSz;
-        WMEMCPY(newKey, in, inSz);
+        long e;
```

Comment:
This should only occur if `!isPrivate`. Make sure the old isPrivate code still returns the `*out = newKey;` etc.